apps/chat/conversation.py

```diff
@@ -158,6 +165,14 @@ def compress_pipeline_chat_history(
     keep_history_len: int = 10,
     history_mode: str = None,
 ) -> list[BaseMessage]:
+    if history_mode == PipelineChatHistoryModes.MAX_HISTORY_LENGTH:
+        limit = keep_history_len // 2 + 1  # each pipeline history message is a pair
```

Comment:
Nit: Should the 1 not be added only when `keep_history_len` is uneven?

Reply:
You should probably also check that the number of returned messages doesn't exceed the token limit